llvm/lib/Target/X86/Disassembler/X86Disassembler.cpp

```diff
@@ -1330,7 +1330,8 @@ static int getInstructionID(struct InternalInstruction *insn,
   //  any position.
   if ((insn->opcodeType == ONEBYTE && ((insn->opcode & 0xFC) == 0xA0)) ||
       (insn->opcodeType == TWOBYTE && (insn->opcode == 0xAE)) ||
-      (insn->opcodeType == THREEBYTE_38 && insn->opcode == 0xF8)) {
+      (insn->opcodeType == THREEBYTE_38 && insn->opcode == 0xF8) ||
+      (insn->vectorExtensionType == TYPE_EVEX && insn->opcode == 0xF8)) {
```

Comment:
Why do the need line 1334? Isn't extended MOVDIR64B's opcodeType THREEBYTE_38?

Reply:
Never mind. I forgot it's in MAP4.

Reply:
Should we add the limitation to MAP4? I think we should have other `insn->opcode == 0xF8` for EVEX encoding.

Reply:
+1, and then remove `insn->vectorExtensionType == TYPE_EVEX` since MAP4 can only be encoded by EVEX.

Reply:
Yes, we do have other F8, will do.